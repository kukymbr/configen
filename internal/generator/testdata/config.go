package testdata

import "errors"

// Added as an example usage.
// To regenerate example files in the configen repository, use `make generate_example`.
//go:generate go tool configen --struct=Config --yaml=true --env=example.env

// Config is an example application config.
type Config struct {
	// App is an application common settings.
	App AppConfig `envPrefix:"APP_" json:"app" yaml:"app"`

	// Logger is a logging setup values.
	Logger LoggerConfig `envPrefix:"LOG_" json:"log" yaml:"logger"`

	// API is an API server configuration.
	API APIConfig `envPrefix:"API_" json:"api" yaml:"api"`
}

type AppConfig struct {
	Env       string `env:"ENV" default:"development" json:"env" yaml:"env"`
	Namespace string `env:"NAMESPACE" default:"unknown" json:"namespace" yaml:"namespace"`
	Domain    string `env:"DOMAIN" example:"example.com" json:"domain" yaml:"domain"`
}

type LoggerConfig struct {
	Level LogLevel `env:"LEVEL" envDefault:"debug" json:"level" yaml:"level"`
}

type APIConfig struct {
	Host string `env:"HOST" envDefault:"0.0.0.0" json:"host" yaml:"host"`
	Port int    `env:"PORT" envDefault:"8080" json:"port" yaml:"port"`
}

type LogLevel int

func (l *LogLevel) MarshalText() ([]byte, error) {
	switch *l {
	case 0:
		return []byte("debug"), nil
	case 1:
		return []byte("info"), nil
	case 2:
		return []byte("error"), nil
	}

	return []byte(""), errors.New("invalid log level")
}
