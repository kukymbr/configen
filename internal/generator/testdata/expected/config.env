# Config is an example application config.

# This file is generated by github.com/kukymbr/configen.
# Source struct: Config

APP_ENV=development
APP_NAMESPACE=unknown
APP_DOMAIN=example.com

LOG_LEVEL=debug

API_HOST=0.0.0.0
API_PORT=8080
