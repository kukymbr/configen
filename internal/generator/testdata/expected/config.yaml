# Config is an example application config.

# This file is generated by github.com/kukymbr/configen.
# Source struct: Config

# App is an application common settings.
app:
    env: development
    namespace: unknown
    domain: example.com
# Logger is a logging setup values.
logger:
    level: debug
# API is an API server configuration.
api:
    host: 0.0.0.0
    port: 8080
